src/app/test/business/dad_jokes_service_test.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:app/business/dad_jokes/dad_jokes_service.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+import '../mocks/dad_jokes_data_mock.dart';
+import '../mocks/dad_jokes_list_mock.dart';
+import 'dad_jokes_service_test.mocks.dart';
+
+@GenerateNiceMocks(
+  [
+    MockSpec<DadJokesRepository>(),
+  ],
+)
+void main() {
+  late DadJokesRepository dadJokesRepository;
+  late FavoriteDadJokesRepository favoriteDadJokesRepository;
+  late DadJokesService SUT;
+
+  setUp(() {
+    TestWidgetsFlutterBinding.ensureInitialized();
+    SharedPreferences.setMockInitialValues({
+      "favoriteDadJokes":
+          getFavoriteDadJokesList().map((dadJoke) => jsonEncode(dadJoke)).toList()
+    });
+
+    provideDummy(dadJokeResponse);
+    dadJokesRepository = MockDadJokesRepository();
+    when(dadJokesRepository.getDadJokes()).thenAnswer(
+      (_) async => dadJokeResponse,
+    );
+
+    favoriteDadJokesRepository = FavoriteDadJokesRepository();
+    SUT = DadJokesService(dadJokesRepository, favoriteDadJokesRepository);
+  });
+
+  test('Get jokes', () async {
+    var result = await SUT.getDadJokes();
+    var mockedJokesList = getDadJokesList().map(
+      (dadJokeChildData) {
+        return DadJoke(
+          id: dadJokeChildData.id,
+          title: dadJokeChildData.title,
+          text: dadJokeChildData.selfText,
+          isFavorite: getFavoriteDadJokesList().any(
+            (favoriteDadJoke) => favoriteDadJoke.id == dadJokeChildData.id,
+          ),
+        );
+      },
+    ).toList();
+
+    expect(result, mockedJokesList);
+  });
+
+  group("favorites", () {
+    test('Get all favorite jokes', () async {
+      var result = await SUT.getFavoriteDadJokes();
+      var mockedJokesList = getFavoriteDadJokesList()
+          .map(
+            (favoriteDadJoke) => DadJoke.fromData(
+              favoriteDadJoke,
+              isFavorite: true,
+            ),
+          )
+          .toList();
+
+      expect(result, mockedJokesList);
+    });
+
+    test('Add a joke to the list of avorites', () async {
+      var dadJoke = const DadJoke(
+        id: '1',
+        title: 'title',
+        text: 'text',
+        isFavorite: false,
+      );
+
+      var expectedJokes =
+          await favoriteDadJokesRepository.getFavoriteDadJokes();
+      expectedJokes.add(
+        DadJokeContentData(
+          id: dadJoke.id,
+          title: dadJoke.title,
+          selfText: dadJoke.text,
+        ),
+      );
+
+      await SUT.addFavoriteDadJoke(dadJoke);
+      var newSavedJokes =
+          await favoriteDadJokesRepository.getFavoriteDadJokes();
+
+      expect(newSavedJokes[newSavedJokes.length - 1].id,
+          expectedJokes[expectedJokes.length - 1].id);
+    });
+
+    test('Remove a favorite joke', () async {
+      var dadJoke = const DadJoke(
```

Comment:
Generally, it's good practice to have content inside DadJoke that are not relevant to the test be random. Typically, on .NET it can be due using `AutoFixture`. I'm not sure if there are equivalent on dart

Reply:
I'm wondering how important this is in a case like this, where the data is not really manipulated?

Reply:
We talked about it and the only options for dart are not well maintained so it will stay like that for now